### ISSUE TITLE
heketi 4.0 rpm missing from EPEL

### ISSUE THREAD

**Issue by sgoodliff** [heketi/heketi#750]
Hi,

where would I find a heketi 4 rpm ?, there is a 3.0.0 version available at https://dl.fedoraproject.org/pub/epel/7/x86_64/ 

or is there  a spec file to build one ?

Thanks

**Comment by jarrpa** [comment]
@lpabon @sgoodliff I'll get to this as soon as I can. However, I'm traveling for the next week, so I can't guarantee how much time I'll be able to put into it during that. In the mean time, the repo containing the spec file is here: http://pkgs.fedoraproject.org/cgit/rpms/heketi.git/tree/?h=epel7 if you do any work towards a heketi 4 RPM, we would certainly appreciate it if you'd contribute it to the project. :) We can use this Issue to keep in contact regarding that development (on either side).

**Comment by sgoodliff** [comment]
Ok thanks, I wont have much time until Thursday next week but ill have a go and let you know.

**Comment by sgoodliff** [comment]
FYI, I've got a v4 rpm built, will be deploying it in a couple of test environments this afternoon ( uk time) to check i've not done something silly.

**Comment by sgoodliff** [comment]
Is there any where i can put the modified spec file and the rpm ?

**Comment by jarrpa** [comment]
@sgoodliff If you have any means of uploading the patch(es), choose your favorite and paste a link here. :) Otherwise send me an email with an attachment, mentioning this PR in the subject line. We'll try to reproduce the binaries.

**Comment by sgoodliff** [comment]
https://gist.github.com/sgoodliff/408c7892e712acffffda638eedb41812

The heketi.service file was missing a -, its correct in https://github.com/heketi/heketi/blob/master/extras/systemd/heketi.service

but not in the repo you sent earlier.

I had to change global_commit as its hard-coded, I will see if i can modify create_deps.sh to fill that in at some point

Hope that helps

**Comment by jarrpa** [comment]
@sgoodliff Good catch.

Can you use the full hash instead of the short hash? Also, Release should be 1 instead of 4, as it is the first release of version 4.

If you want, I'd encourage you to create a full commit with a short commit message and upload the output of "git show HEAD" so you get proper credit. :)

**Comment by sgoodliff** [comment]
something like this https://gist.github.com/sgoodliff/9588ae2ebc52a5096e44b1f6980398a6 ?

Thanks

**Comment by jarrpa** [comment]
@sgoodliff Indeed! If no one else gets to it beforehand, I'll work on producing official RPMs on Monday.

**Comment by MohamedAshiqrh** [comment]
@jarrpa Any update on this? Can you also teach me to build it?

**Comment by sgoodliff** [comment]
I'd be happy to build a new rpm or update the spec file every release if there was somewhere useful I could push it to

**Comment by jarrpa** [comment]
@sgoodliff @MohamedAshiqrh Apologies for the long silence. I managed to push a build of heketi 4 to EPEL a few days ago, but it has a few more days of sitting in testing to go. If you can test out the RPMs and verify that they work, please give the build a +1 here: https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2017-373830f419

**Comment by sgoodliff** [comment]
Hi,

I just downloaded it and ran 

rpm2cpio heketi-4.0.0-1.el7.x86_64.rpm | cpio -idmv

the service seems to be missing my fix for the ExecStart line.

/usr/lib/systemd/system/heketi.service

ExecStart=/usr/bin/heketi -config=/etc/heketi/heketi.json

**Comment by jarrpa** [comment]
Oh no! I'm sorry, I thought I got that. :( I'll update it.

**Comment by sgoodliff** [comment]
no worries, let me know when its ready and ill grab it and push it to a test environment.

thanks

**Comment by jarrpa** [comment]
....well I'm dumb. I checked my code and the change is there for that version. But I also checked the related build in koji and found this:

https://koji.fedoraproject.org/koji/fileinfo?rpmID=9817343&filename=heketi.service

Which has a Digest that doesn't match the file in my repo but DOES match the old version of the file without the fix. So then I check the git history...

...and I forgot to commit and push that change. D'OH!!

I'll be sending out a new build shortly. :)
